### PR TITLE
Pu/matrixinput

### PR DESCRIPTION
**Pull request by izulin**
### Context

Adds parsing of arrays.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1.
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [x] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

File: src/interpreter/Interpreter.ts [hunk 1]
```diff
@@ -226,6 +228,38 @@ export class Interpreter {
...
+            const height = arr.height()
+            if(rowHeight===undefined) {
+              rowHeight = height
+              rowRet.push(...arr.data)
+            } else if(rowHeight === height) {
+              for(let i=0;i<height;i++) {
+                rowRet[i].push(...arr.data[i])
+              }
+            } else {
+              return new CellError(ErrorType.REF, ErrorMessage.SizeMismatch)
```
Comment (voodoo11) [inline src/interpreter/Interpreter.ts:248]: I would say it's `ErrorType.VALUE`

Comment (izulin) [reply src/interpreter/Interpreter.ts:248]: Screenshot 2021-05-19 at 10 44 31

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]

File: src/Config.ts [hunk 2]
```diff
@@ -611,7 +635,12 @@ export class Config implements ConfigParams, ParserConfig {
...
       {value: this.decimalSeparator, name: 'decimalSeparator'},
       {value: this.functionArgSeparator, name: 'functionArgSeparator'},
-      {value: this.thousandSeparator, name: 'thousandSeparator'}
+      {value: this.thousandSeparator, name: 'thousandSeparator'},
+    )
+
+    configCheckIfParametersNotInConflict(
+      {value: this.matrixRowSeparator, name: 'matrixRowSeparator'},
+      {value: this.matrixColumnSeparator, name: 'matrixColumnSeparator'},
     )
```
[the remark this answers is not in the document]
Comment (izulin) [reply src/Config.ts:644]: parser will handle this

[the remark this answers is not in the document]
Comment (izulin) [reply src/Config.ts:643]: as in: parser already handles well the conflict those have over ',' symbol

[the remark this answers is not in the document]
Comment (wojciechczerniak) [reply src/Config.ts:643]: 👍

**Review by wojciechczerniak — APPROVED** [review]
🥇

**Pull request merged**